From b0987eeed829ba1986f84171f6c13c4bb3399fa4 Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Thu, 14 Nov 2024 16:17:00 +0100
Subject: [PATCH 1383/1695] dts: stm32: add OEM rproc encryption key for
 stm32mp21

Add new OTP entry for remoteproc encryption key.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 262d47fac3a0e7fae42f54e69d1b988749934d85 dts: stm32: add OEM rproc encryption key for stm32mp21]
Change-Id: I4cd89516de752bbc703cd5bcf2b716b0643d152b
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/418746
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 522ce1508..55dc3aecc 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -2459,6 +2459,12 @@
 				st,non-secure-otp-provisioning;
 			};
 
+			oem_rproc_enc_key: oem_rproc_enc_key@530 {
+				/* 0x530 -> otp332 */
+				reg = <0x530 0x20>;
+				st,non-secure-otp-provisioning;
+			};
+
 			oem_enc_key_2: otp348@570 {
 				reg = <0x570 0x20>;
 				st,non-secure-otp-provisioning;
-- 
2.47.2

